src/lib/paste.ts

```diff
@@ -138,6 +138,7 @@ export const handlePaste = (
 
   if (
     selection instanceof TextSelection &&
+    selection.$from.depth === slice.openStart &&
```

Comment:
I'm not sure if that's a valid restriction to allow to insert slices that are matching selection depth. Wouldn't it be a weird behaviour from user's point of view? Should we maybe find a closest position where a block element can be inserted? Or maybe we need to split the text before we insert a block element here? In general it just looks like tr.replace has too much work to do.

Reply:
updated to decide that from slice depth